ludwig/benchmarking/utils.py

```diff
@@ -0,0 +1,125 @@
+import copy
+import logging
+import os
+import shutil
+from datetime import datetime
+from types import ModuleType
+from typing import Any, Dict, Tuple, Union
+
+import fsspec
+import pandas as pd
+from botocore.exceptions import ClientError
+from globals import CONFIG_YAML, EXPERIMENT_RUN, REPORT_JSON
+from s3fs.errors import translate_boto_error
+
+from ludwig.constants import CATEGORY
+from ludwig.globals import MODEL_HYPERPARAMETERS_FILE_NAME
+from ludwig.utils.dataset_utils import get_repeatable_train_val_test_split
+from ludwig.utils.defaults import default_random_seed, merge_with_defaults
+from ludwig.utils.fs_utils import get_fs_and_path
+
+
+def get_full_config(config: Dict[str, Any]) -> Dict[str, Any]:
+    """Return a full Ludwig config given an experiment config.
+
+    config: ludwig config.
+    """
+    config_dict = copy.deepcopy(config)
+    config = merge_with_defaults(config_dict)
+    return config
+
+
+def load_from_module(
+    dataset_module: ModuleType, output_feature: Dict[str, str], subsample_frac: float = 1
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Take in Ludwig Python module of the dataset, load it, and split it if it isn't already (by adding "split"
+    column).
+
+    dataset_module: ludwig dataset module (e.g. ludwig.datasets.sst2, ludwig.datasets.ames_housing, etc.)
+    subsample_frac: percentage of the total datasets to load.
+    """
+    dataset = dataset_module.load(split=False)
+    if subsample_frac < 1:
+        dataset = dataset.sample(frac=subsample_frac, replace=False, random_state=default_random_seed)
+
+    if output_feature["type"] == CATEGORY:
+        dataset = get_repeatable_train_val_test_split(
+            dataset,
+            stratify_colname=output_feature["name"],
+            random_seed=default_random_seed,
+            frac_train=0.75,
+            frac_val=0.1,
+            frac_test=0.15,
+        )
+    else:
+        dataset = get_repeatable_train_val_test_split(
+            dataset, random_seed=default_random_seed, frac_train=0.75, frac_val=0.1, frac_test=0.15
+        )
+
+    train_df = dataset[dataset["split"] == 0]
+    val_df = dataset[dataset["split"] == 1]
+    test_df = dataset[dataset["split"] == 2]
+    return train_df, val_df, test_df, dataset
+
+
+def bytes_to_human_readable(num: Union[float, int], suffix: str = "B") -> str:
+    """Transform bytes into a human readable format."""
+    for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
+        if abs(num) < 1024.0:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1024.0
+    return f"{num:.1f}Yi{suffix}"
+
+
+def flatten_dict(d: Dict[str, Any], sep: str = ".") -> Dict[str, Any]:
+    [flat_dict] = pd.json_normalize(d, sep=sep).to_dict(orient="records")
+    return flat_dict
+
+
+def export_artifacts(
+    experiment: Dict[str, str], report_path: str, experiment_output_directory: str, export_base_path: str
```

Comment:
What are the two strings in the `experiment` argument?

Reply:
Added to the docstring:
```
experiment: experiment dict that contains "dataset_name" (e.g. ames_housing),
        "experiment_name" (specified by user), and "config_path" (path to experiment config.
        Relative to ludwig/benchmarks/configs).
```